defaults.js

```diff
@@ -25,7 +25,11 @@ export const getArgs = (args) => {
 	const convertedArgs = {};
 	Object.keys(args).forEach((a) => {
 		if (!!openAiToLlamaMapping[a]) {
-			convertedArgs[openAiToLlamaMapping[a]] = args[a];
+			if (a === 'max_tokens' && args[a] === null) {
+				convertedArgs[openAiToLlamaMapping[a]] = -1;
+			} else {
+				convertedArgs[openAiToLlamaMapping[a]] = args[a];
+			}
```

Comment:
if we want the default to be -1 we should just keep the original code the same here and update the defaultParams for --n_predict to be -1.

Reply:
I don't have time this morning to look at all of the code, but the behavior of the OpenAI API is to use the default (in this case -1 for ChatCompletion) if a null is explicitly passed. So there's some trickiness in preserving that quirk. It probably applies to every other option passed via the API that has a default.

In the client that I was attempting to fix, unrelated to this project, I came to the realization last night that it was incorrectly using the max_token to determine what is actually in the context (to limit the number of messages sent on each ChatCompletion) versus using a lookup table depending on the model (different models have different context limits.) So in the client I'm talking about, again unrelated to this project, there's more fixes to be made.

I'll rebase and fix according to your feedback as soon as I get a chance if you are interested in preserving this quirk of the OpenAI API, in your project. Thank you!

Reply:
sounds good! i'll be on the lookout!